### ISSUE TITLE
Print/Accept memory in human readable format

### ISSUE THREAD

**Issue by k-sareen** [mmtk/mmtk-core#829]
Debug builds print memory like so:
```console
[INFO  mmtk::memory_manager] Initialized MMTk with NoGC (FixedHeapSize(16803151872))
```
I think it'd be nicer if the memory was printed in a human readable format. So something like
```console
[INFO  mmtk::memory_manager] Initialized MMTk with NoGC (FixedHeapSize(16 G))
```

A related issue is we have to provide memory in bytes usually (for example if setting the nursery size). It would be more convenient if we also accepted human readable format for those options.

**Comment by wks** [comment]
I agree with the part that we should probably accept human readable format for those options.

But when printing, I still believe the information should be accurate, i.e. humans see what programs see.  16803151872 is definitely more accurate than 16 G, and the program may behave differently depending on whether the heap size is 16803151872 bytes or 16000000000 bytes or 16999999999 bytes.  By having byte (or page) precision, the programmer can identify that two logs are from programs running at different heap sizes.

For dynamic heap sizes, a 1% change in heap size will not be shown in "16G".  I think sometimes it should be useful to see tiny changes in heap size that that because that may be useful to explain the behaviour or MemBalancer in corner cases.

It should be good for both human and machine to divide the heap size by thousands separators, such as 16_803_151_872 (or 168_0315_1872 if you are Chinese or Japanese).

**Comment by k-sareen** [comment]
The smart way to do this would be to print accurate info into log files but print human readable on the console. But maybe that's not intuitive to users. I agree that it is nice to have accurate information
